contracts/protocols/mainnet/uniswap/helpers.sol

```diff
@@ -0,0 +1,351 @@
+    }
+
+    function convertTo18(uint256 _dec, uint256 _amt) internal pure returns (uint256 amt) {
+        amt = mul(_amt, 10**(18 - _dec));
+    }
+
+    function changeEthAddress(address buy, address sell)
+        internal
+        pure
+        returns (TokenInterface _buy, TokenInterface _sell)
+    {
+        _buy = buy == ethAddr ? TokenInterface(wethAddr) : TokenInterface(buy);
+        _sell = sell == ethAddr ? TokenInterface(wethAddr) : TokenInterface(sell);
+    }
+
+    /**
+     * @dev Return uniswap v3 Swap Router Address
+     */
+    function getUniswapRouterAddr() internal pure returns (address) {
+        return 0xE592427A0AEce92De3Edee1F18E0157C05861564;
+    }
+
+    /**
+     * @dev Get Last NFT Index
+     * @param user: User address
+     */
+    function getLastNftId(address user) internal view returns (uint256 tokenId) {
+        uint256 len = nftManager.balanceOf(user);
+        tokenId = nftManager.tokenOfOwnerByIndex(user, len - 1);
+    }
+
+    function getMinAmount(
+        TokenInterface token,
+        uint256 amt,
+        uint256 slippage
+    ) internal view returns (uint256 minAmt) {
+        uint256 _amt18 = convertTo18(token.decimals(), amt);
+        minAmt = wmul(_amt18, sub(WAD, slippage));
+        minAmt = convert18ToDec(token.decimals(), minAmt);
+    }
+
+    function positions(uint256 tokenId)
+        internal
+        view
+        returns (
+            address token0,
+            address token1,
+            uint24 fee,
+            int24 tickLower,
+            int24 tickUpper,
+            uint128 liquidity
+        )
+    {
+        (, , token0, token1, fee, tickLower, tickUpper, liquidity, , , , ) = nftManager.positions(tokenId);
+    }
+
+    function getPoolAddress(
+        address token0,
+        address token1,
+        uint24 fee
+    ) internal view returns (address poolAddr) {
+        poolAddr = PoolAddress.computeAddress(
+            nftManager.factory(),
+            PoolAddress.PoolKey({ token0: token0, token1: token1, fee: fee })
+        );
+    }
+
+    function depositAmount(
+        uint256 tokenId,
+        uint256 amountA,
+        uint256 amountB
+    ) internal view returns (uint256 liquidity) {
+        (, , address _token0, address _token1, uint24 _fee, int24 tickLower, int24 tickUpper, , , , , ) = nftManager
+        .positions(tokenId);
+
+        IUniswapV3Pool pool = IUniswapV3Pool(getPoolAddress(_token0, _token1, _fee));
+
+        // compute the liquidity amount
+        {
+            (uint160 sqrtPriceX96, , , , , , ) = pool.slot0();
+            uint160 sqrtRatioAX96 = TickMath.getSqrtRatioAtTick(tickLower);
+            uint160 sqrtRatioBX96 = TickMath.getSqrtRatioAtTick(tickUpper);
+
+            liquidity = LiquidityAmounts.getLiquidityForAmounts(
+                sqrtPriceX96,
+                sqrtRatioAX96,
+                sqrtRatioBX96,
+                amountA,
+                amountB
+            );
+        }
```

Comment:
@cryptoDev222 After this, we need to check at what ratio amountA and amountB is required. So, we need to pass `liquidity` into this function: https://github.com/Uniswap/uniswap-v3-periphery/blob/1f5b385675eee30a77ee10f4dfd6aa0aaa766026/contracts/libraries/LiquidityAmounts.sol#L120

To get `amountA` and `amountB`

Reply:
You mean something like this.
https://github.com/Instadapp/dsa-resolvers-old/blob/master/contracts/protocols/mainnet/uniswapV2.sol#L356